templates/workloads/services/lb-web/manifest.yml

```diff
@@ -16,10 +16,17 @@ image:
 http:
   # Requests to this path will be forwarded to your service. 
   # To match all requests you can use the "/" path. 
+  # NOTE: If you plan to deploy multiple services, this value must be unique per-service
+  # NOTE: It is unnecessary to start the path with a "/", for example "/api" versus "api"
   path: '{{.Path}}'
   # You can specify a custom health check path. The default is "/"
+  # WARNING: Passing healthchecks are required and are how ECS measures task health. Failure to pass healthcheck may result cyclical restarts.
+  # NOTE: The healthcheck URL should be appended to the "path" value.
+  # For example, if "path" is set to "api", and your API serves it's healthcheck on "/healthz", this value should be "/api/healthz"
   # healthcheck: '{{.HealthCheckPath}}'
```

Comment:
```suggestion
  # healthcheck: '{{.HealthCheckPath}}' {{if ne .Path "/"}} # Must be a path under '{{.Path}}'. For example: '{{.Path}}/_healthcheck' {{end}}
```

Reply:
I double checked on this one -- for one service, it threw errors that the prefixed path wasn't found.

Think this one has a weird interplay with the Load Balancer. For example, with Hasura:
https://github.com/hasura/graphql-engine

It seems that if you put this at a subpath, `path: "hasura"`, you cannot load the web console (`/console`) after it being deployed with Copilot. 

Because the root path already had `/hasura` in it, when I tried to visit `/hasura/console`, it throw nonexistent path error from the Hasura service, and then in the Fargate logs I could see it wasn't stripping the `/hasura` prefix from the route:

![image](https://user-images.githubusercontent.com/26604994/93026112-941b2b80-f5d1-11ea-9e33-acdb5c8acb7e.png)

See `http_info.url` here:
```json
{
  "type": "http-log",
  "timestamp": "2020-09-13T18:55:04.819+0000",
  "level": "error",
  "detail": {
    "operation": {
      "error": {
        "path": "$",
        "error": "resource does not exist",
        "code": "not-found"
      },
      "request_id": "105a938a-82b6-4b62-83bc-594ddaa2fc73",
      "response_size": 65,
      "raw_query": ""
    },
    "http_info": {
      "status": 404,
      "http_version": "HTTP/1.1",
      "url": "/hasura/console",
      "ip": "73.179.121.23",
      "method": "GET",
      "content_encoding": null
    }
  }
}
```

In Hasura's case, the Healthcheck endpoint lives at `/healthz`, but giving `/hasura/healthz` as a path to Copilot results in failure, and giving `/healthz` cannot work because it'll never hit the `/hasura` path and be routed :grimacing: 

![image](https://user-images.githubusercontent.com/26604994/93026249-a21d7c00-f5d2-11ea-8588-750c1658990b.png)


This kind of effectively renders the web service useless since none of the paths will route properly.

(Unrelated: Would attaching a domain fix this?)

Might be better for someone from your team who understands this more to weigh in here, I think I may have been mistaken on this one, sorry.

Reply:
Heya! 

I double-checked this and the healthcheck path should still be the path in the service code (just `/healthz` and not `/hasura/healthz`). I believe this is because it's a property of the Target Group so you don't need to prefix it with the rule path ("hasura").

Out of curiosity: Does running the container locally work? My suspicion is that the container fails to come up so it never gets to the point where it can query the "/healthz" path.

Reply:
Hmm :thinking:

The `hasura` instance does run locally, and it was running on the ELB there (that error format with `path: $, error: xxx` is Hasura's web server response) but because of the LB having it at `/hasura`, when Hasura router handled the requests it was trying to send them to wrong place.

For example:
```
Load Balanced Route | Desired App Path | Routed App Path               
----------------------------------------------------------
/hasura/console     |     /console     | /hasura/console
```

So basically because the app webserver router is parsing `/hasura/console` and not stripping `/hasura`, it won't route properly.

> healthcheck path should still be the path in the service code (just /healthz and not /hasura/healthz). I believe this is because it's a property of the Target Group so you don't need to prefix it with the rule path ("hasura").

This makes sense -- so it is not possible to check the `healthcheck` path from outside of the running container then right?